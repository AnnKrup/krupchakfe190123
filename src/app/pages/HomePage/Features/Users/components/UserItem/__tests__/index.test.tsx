import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { UserItem } from '../index';
import { fireEvent, render } from '@testing-library/react';

const renderer = createRenderer();

const props = {
  user: {
    phone: '902-782-3286',
    name: 'Brazil Izair',
    nickname: '@brazilizair',
    email: 'brazil.izair@fallinhay.com',
    position: 'Managing Director',
    photo: '2.jpg',
  },
  chooseUser: () => {},
};

describe('<UserItem />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<UserItem {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('check image', () => {
    const { getByAltText } = render(<UserItem {...props} />);
    getByAltText(`user ${props.user.name}`);
  });

  it('choose user fires', () => {
    const chooseUser = jest.fn();
    const { getByText } = render(
      <UserItem {...props} chooseUser={chooseUser} />,
    );
    const button = getByText(`View`);

    fireEvent.click(button);

    expect(chooseUser).toHaveBeenCalled();
  });
});
