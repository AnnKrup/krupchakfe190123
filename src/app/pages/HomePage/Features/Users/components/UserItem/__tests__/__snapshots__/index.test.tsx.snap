// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> should render and match the snapshot 1`] = `
<div
  className="UserItem"
>
  <div
    className="UserItemInfo"
  >
    <img
      alt="user Brazil Izair"
      className="UserItemIcon"
      src="foto/2.jpg"
    />
    <div>
      <div>
        Brazil Izair
      </div>
      <div>
        @brazilizair
      </div>
    </div>
  </div>
  <div
    className="UserItemBtnView"
    onClick={[Function]}
  >
    View
  </div>
</div>
`;
